pilot/pkg/serviceregistry/external/servicediscovery.go

```diff
@@ -95,6 +95,9 @@ func NewServiceDiscovery(configController model.ConfigStoreCache, store model.Is
 }
 
 // getWorkloadEntryHandler defines the handler for workload entries
+// kube registry controller also calls this function indirectly via the Share interface
```

Comment:
Does it have to be kube registry ? It seems that now it's a pretty generic implementation - Consul or anything else would also work, if they generate same events ?

Reply:
sure they can but right now, its only kube